fdb-record-layer-core/src/main/proto/record_query_plan.proto

```diff
@@ -1724,3 +1724,13 @@ message PRecordQueryUnionPlanBase {
 message PRecordQueryUpdatePlan {
   optional PRecordQueryAbstractDataModificationPlan super = 1;
 }
+
+//
+// PTypedQueryResult
+//
+message PQueryResult {
+  oneof datum {
+    PComparableObject primitive = 1;
+    bytes complex = 2;
```

Comment:
only the `datum` is serialized. it can be:
- either a primitive type, for that I am reusing `PComparableObject` and its SerDe APIs, instead of more or less duplicating it.
- or a complex type, i.e. a record represented as a `Message`, that is serialized as-is.
Maybe it is better to introduce a different, more elaborate, PB message for the serialization of `datum` that does not reuse `PComparableObject`.